src/main/java/com/aws/greengrass/shadowmanager/configuration/ComponentConfiguration.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.configuration;
+
+import com.aws.greengrass.config.Topics;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+
+public class ComponentConfiguration {
+    RateLimitsConfiguration rateLimitsConfiguration;
+
+    public ComponentConfiguration(RateLimitsConfiguration rateLimitsConfiguration) {
```

Comment:
```suggestion
    private ComponentConfiguration(RateLimitsConfiguration rateLimitsConfiguration) {
```

Otherwise people can skip the creation logic on the factory method to properly create an instance of the configuration